src/cad-pulse.c

```diff
@@ -577,13 +577,23 @@ static void init_card_info(pa_context *ctx, const pa_card_info *info, int eol, v
     for (i = 0; i < info->n_ports; i++) {
         pa_card_port_info *port = info->ports[i];
 
+#ifdef WITH_DROID_SUPPORT
+        if (strstr(port->name, DROID_OUTPUT_PORT_SPEAKER) != NULL) {
+            has_speaker = TRUE;
+        } else if (strstr(port->name, DROID_OUTPUT_PORT_EARPIECE) != NULL ||
+                   strstr(port->name, DROID_OUTPUT_PORT_WIRED_HEADSET)  != NULL) {
+            has_earpiece = TRUE;
+        }
+    }
+#else
```

Comment:
```suggestion
#endif /* WITH_DROID_SUPPORT */
```

The rest of droid-specific changes try not to change behaviour with upstream callaudiod. I think it's safe to add this as a separate block so that the check for `SND_USE_CASE_DEV_{SPEAKER,EARPIECE,HANDSET}` still works as a fallback.